protocol/test/forge/v2/interestRate/CompoundedInterest.t.sol

```diff
@@ -0,0 +1,89 @@
+  uint256 public constant FIVE_PCT_30DAY = 100_411804498165141800;
+  uint256 public constant FIVE_PCT_1YEAR = 105_127109637602403900;
+  
+  uint256 public constant TEN_PCT_INTEREST = 0.10e18;
+  uint256 public constant TEN_PCT_1DAY = 100_027401013666092800;
+  uint256 public constant TEN_PCT_30DAY = 100_825304825777374100;
+  uint256 public constant TEN_PCT_1YEAR = 110_517091807564762400;
+
+
+  uint256 public initialPrincipalAmount = 100e18;
+  
+}
+
+contract CompoundedInterestTestModifiers is CompoundedInterestTestBase {
+
+  // One percent interest tests
+  function test_computeInterestRateOnePctOneDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 1 days, uint96(ONE_PCT_INTEREST));
+    assertEq(newInterestRate, ONE_PCT_1DAY);
+  }
+  
+  function test_computeInterestRateOnePctThirtyDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 30 days, uint96(ONE_PCT_INTEREST));
+    assertEq(newInterestRate, ONE_PCT_30DAY);
+  }
+  
+  function test_computeInterestRateOnePctOneYearImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 365 days, uint96(ONE_PCT_INTEREST));
+    assertEq(newInterestRate, ONE_PCT_1YEAR);
+  }
+
+  // Five percent interest tests
+  function test_computeInterestRateFivePctOneDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 1 days, uint96(FIVE_PCT_INTEREST));
+    assertEq(newInterestRate, FIVE_PCT_1DAY);
+  }
+  
+  function test_computeInterestRateFivePctThirtyDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 30 days, uint96(FIVE_PCT_INTEREST));
+    assertEq(newInterestRate, FIVE_PCT_30DAY);
+  }
+  
+  function test_computeInterestRateFivePctOneYearImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 365 days, uint96(FIVE_PCT_INTEREST));
+    assertEq(newInterestRate, FIVE_PCT_1YEAR);
+  }
+
+  // Ten percent interest tests
+  function test_computeInterestRateTenPctOneDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 1 days, uint96(TEN_PCT_INTEREST));
+    assertEq(newInterestRate, TEN_PCT_1DAY);
+  }
+  
+  function test_computeInterestRateTenPctThirtyDayImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 30 days, uint96(TEN_PCT_INTEREST));
+    assertEq(newInterestRate, TEN_PCT_30DAY);
+  }
+  
+  function test_computeInterestRateTenPctOneYearImpl() public {    
+    uint256 newInterestRate = CompoundedInterest.continuouslyCompounded(initialPrincipalAmount, 365 days, uint96(TEN_PCT_INTEREST));
+    assertEq(newInterestRate, TEN_PCT_1YEAR);
+  }
+
```

Comment:
Can you make sure we have the other edge cases listed? What about 0 days, 0 principal, 0 interest rate?

Any others you can think of?